### PR TITLE
support for rook Device.FullPath

### LINKED ISSUE #3374: Disk discovery need to accept device by path
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**

Currently, we take kernel device name as enumerated by the kernel. While this is easy, it is not safe. Name needs to be autoselected based on full path to one of dev link specified.

**What is use case behind this feature:**

Extend rook Device object with FullPath support.

**Environment**:

### PR DESCRIPTION
**Pull request by dyusupov**
Signed-off-by: Dmitry Yusupov <dmitry.yusupov@nexenta.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #3374 

**Checklist:**
- [x] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../CONTRIBUTING.md#comments)

// known CI issues
[skip ci]

### CODE REVIEW

File: Documentation/ceph-cluster-crd.md [hunk 1]
```diff
@@ -138,6 +138,7 @@ Below are the settings available, both at the cluster and individual node level,
   - `^[^r]`: Selects all devices that do *not* start with `r`
 - `devices`: A list of individual device names belonging to this node to include in the storage cluster.
   - `name`: The name of the device (e.g., `sda`).
+  - `fullpath`: The full path to the device (e.g., `/dev/disk/by-id/scsi-35000c5008335c83f`). If specified then `name` can be omitted.
```
Comment (travisn) [inline Documentation/ceph-cluster-crd.md]: I'd suggest to leave out the ceph changes with this PR. Ceph will require more in-depth changes as seen in #2845. Let's go ahead with this to get it working for edgefs, then we should get the ceph PR rebased on this.

File: pkg/operator/discover/discover.go [hunk 2]
```diff
@@ -298,6 +299,16 @@ func ListDevicesInUse(context *clusterd.Context, namespace, nodeName string) ([]
 	return devices, nil
 }
 
+func matchFullPath(dls, fp string) bool {
```
Comment (travisn) [inline pkg/operator/discover/discover.go]: would you mind either renaming the variables to be descriptive or add comments about dls and fp?

File: pkg/operator/edgefs/cluster/target/layout.go [hunk 3]
```diff
@@ -56,7 +56,7 @@ func getIdDevLinkName(dls string) (dl string) {
 	dlsArr := strings.Split(dls, " ")
 	for i := range dlsArr {
 		s := strings.Replace(dlsArr[i], "/dev/disk/by-id/", "", 1)
-		if strings.Contains(s, "/") || strings.Contains(s, "wwn-") {
+		if strings.Contains(s, "/") || strings.Contains(s, "wwn-") || strings.Contains(s, "nvme-nvme.") || strings.Contains(s, "nvme-eui.") {
```
Comment (travisn) [inline pkg/operator/edgefs/cluster/target/layout.go]: the devices specified by fullPath won't be skipped by the check for `/`?

File: Documentation/ceph-cluster-crd.md (hunk 1 above)
Comment (dyusupov) [reply Documentation/ceph-cluster-crd.md]: done

File: pkg/operator/discover/discover.go (hunk 2 above)
Comment (dyusupov) [reply pkg/operator/discover/discover.go]: done

File: pkg/operator/edgefs/cluster/target/layout.go (hunk 3 above)
Comment (dyusupov) [reply pkg/operator/edgefs/cluster/target/layout.go]: done

**Pull request merged**